libraries/AP_Common/AP_Common.h

```diff
@@ -44,6 +44,18 @@
 #define ToRad(x) radians(x)	// *pi/180
 #define ToDeg(x) degrees(x)	// *180/pi
 
+/* Declare and implement const and non-const versions of the array subscript
+ * operator. The object is treated as an array of type_ values. */
+#define ARRAY_SUBSCRIPT(type_) \
+inline type_ &operator[](size_t i) \
```

Comment:
The only reasonable type is uint8_t. All the others have problems of increasing required alignment. So... just remove the param, at which point I'd say it's reasonable to just define the methods in place rather than adding the macro.